Docker build fails on installing pydantic-core [linux/arm/v7]

Docker build fails on installing pydantic-core / see report from CI:

- https://github.com/searxng/searxng/actions/runs/11191050608/job/31113678686#step:8:981



```
#24 [linux/arm/v7 5/8] RUN apk add --no-cache -t build-dependencies     build-base     py3-setuptools     python3-dev     libffi-dev     libxslt-dev     libxml2-dev     openssl-dev     tar     git  && apk add --no-cache     ca-certificates     python3     py3-pip     libxml2     libxslt     openssl     tini     uwsgi     uwsgi-python3     brotli  && pip3 install --break-system-packages --no-cache -r requirements.txt  && apk del build-dependencies  && rm -rf /root/.cache
...
#24 221.9 Collecting pydantic-core==2.20.1 (from pydantic==2.8.2->-r requirements.txt (line 19))
#24 221.9   Downloading pydantic_core-2.20.1.tar.gz (388 kB)
#24 222.0      ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 388.4/388.4 kB 18.9 MB/s eta 0:00:00
#24 222.3   Installing build dependencies: started
#24 230.0   Installing build dependencies: finished with status 'done'
#24 230.0   Getting requirements to build wheel: started
#24 230.6   Getting requirements to build wheel: finished with status 'done'
#24 230.6   Preparing metadata (pyproject.toml): started
#24 231.2   Preparing metadata (pyproject.toml): finished with status 'error'
#24 231.2   error: subprocess-exited-with-error
#24 231.2   
#24 231.2   × Preparing metadata (pyproject.toml) did not run successfully.
#24 231.2   │ exit code: 1
#24 231.2   ╰─> [6 lines of output]
#24 231.2       
#24 231.2       Cargo, the Rust package manager, is not installed or is not on PATH.
#24 231.2       This package requires Rust and Cargo to compile extensions. Install it through
#24 231.2       the system's package manager or via https://rustup.rs/
#24 231.2       
#24 231.2       Checking for Rust toolchain....
#24 231.2       [end of output]
#24 231.2   
#24 231.2   note: This error originates from a subprocess, and is likely not a problem with pip.
#24 231.2 error: metadata-generation-failed
#24 231.2 
#24 231.2 × Encountered error while generating package metadata.
#24 231.2 ╰─> See above for output.
#24 231.2 
#24 231.2 note: This is an issue with the package mentioned above, not pip.
#24 231.2 hint: See above for details.
#24 ERROR: process "/bin/sh -c apk add --no-cache -t build-dependencies     build-base     py3-setuptools     python3-dev     libffi-dev     libxslt-dev     libxml2-dev     openssl-dev     tar     git  && apk add --no-cache     ca-certificates     python3     py3-pip     libxml2     libxslt     openssl     tini     uwsgi     uwsgi-python3     brotli  && pip3 install --break-system-packages --no-cache -r requirements.txt  && apk del build-dependencies  && rm -rf /root/.cache" did not complete successfully: exit code: 1
------
 > [linux/arm/v7 5/8] RUN apk add --no-cache -t build-dependencies     build-base     py3-setuptools     python3-dev     libffi-dev     libxslt-dev     libxml2-dev     openssl-dev     tar     git  && apk add --no-cache     ca-certificates     python3     py3-pip     libxml2     libxslt     openssl     tini     uwsgi     uwsgi-python3     brotli  && pip3 install --break-system-packages --no-cache -r requirements.txt  && apk del build-dependencies  && rm -rf /root/.cache:
231.2       [end of output]
231.2   
231.2   note: This error originates from a subprocess, and is likely not a problem with pip.
231.2 error: metadata-generation-failed
231.2 
231.2 × Encountered error while generating package metadata.
231.2 ╰─> See above for output.
231.2 
231.2 note: This is an issue with the package mentioned above, not pip.
231.2 hint: See above for details.
------

 1 warning found (use docker --debug to expand):
 - SecretsUsedInArgOrEnv: Do not use ARG or ENV instructions for sensitive data (ENV "MORTY_KEY") (line 12)
Dockerfile:26
--------------------
  25 |     
  26 | >>> RUN apk add --no-cache -t build-dependencies \
  27 | >>>     build-base \
  28 | >>>     py3-setuptools \
  29 | >>>     python3-dev \
  30 | >>>     libffi-dev \
  31 | >>>     libxslt-dev \
  32 | >>>     libxml2-dev \
  33 | >>>     openssl-dev \
  34 | >>>     tar \
  35 | >>>     git \
  36 | >>>  && apk add --no-cache \
  37 | >>>     ca-certificates \
  38 | >>>     python3 \
  39 | >>>     py3-pip \
  40 | >>>     libxml2 \
  41 | >>>     libxslt \
  42 | >>>     openssl \
  43 | >>>     tini \
  44 | >>>     uwsgi \
  45 | >>>     uwsgi-python3 \
  46 | >>>     brotli \
  47 | >>>  && pip3 install --break-system-packages --no-cache -r requirements.txt \
  48 | >>>  && apk del build-dependencies \
  49 | >>>  && rm -rf /root/.cache
  50 |     
--------------------
ERROR: failed to solve: process "/bin/sh -c apk add --no-cache -t build-dependencies     build-base     py3-setuptools     python3-dev     libffi-dev     libxslt-dev     libxml2-dev     openssl-dev     tar     git  && apk add --no-cache     ca-certificates     python3     py3-pip     libxml2     libxslt     openssl     tini     uwsgi     uwsgi-python3     brotli  && pip3 install --break-system-packages --no-cache -r requirements.txt  && apk del build-dependencies  && rm -rf /root/.cache" did not complete successfully: exit code: 1
ERROR: docker.build exit with error (1)
make: *** [Makefile:94: docker.buildx] Error 1
Error: Process completed with exit code 2.
```
--------

@vojkovic pointed out wheel `pydantic_core-2.20.1-cp313-cp313-manylinux_2_17_armv7l.manylinux2014_armv7l.whl` at [pydantic-core/2.20.1/#files](https://pypi.org/project/pydantic-core/2.20.1/#files)

--- a/Dockerfile
+++ b/Dockerfile
@@ -44,6 +44,7 @@ RUN apk add --no-cache -t build-dependencies \
     uwsgi \
     uwsgi-python3 \
     brotli \
+    py3-pydantic-core \
  && pip3 install --break-system-packages --no-cache -r requirements.txt \
  && apk del build-dependencies \
  && rm -rf /root/.cache

